activerecord/lib/active_record/railties/databases.rake

```diff
@@ -501,6 +545,61 @@ db_namespace = namespace :db do
         db_namespace["test:load"].invoke
       end
     end
+
+    ActiveRecord::Tasks::DatabaseTasks.for_each(databases) do |spec_name|
+      # desc "Recreate the #{spec_name} test database"
+      namespace :load do
+        task spec_name => "db:test:purge:#{spec_name}" do
+          case ActiveRecord::Base.schema_format
+          when :ruby
+            db_namespace["test:load_schema:#{spec_name}"].invoke
+          when :sql
+            db_namespace["test:load_structure:#{spec_name}"].invoke
+          end
+        end
+      end
+
+      # desc "Recreate the #{spec_name} test database from an existent schema.rb file"
+      namespace :load_schema do
+        task spec_name => "db:test:purge:#{spec_name}" do
+          should_reconnect = ActiveRecord::Base.connection_pool.active_connection?
+          ActiveRecord::Schema.verbose = false
+          filename = ActiveRecord::Tasks::DatabaseTasks.dump_filename(spec_name, :ruby)
+          db_config = ActiveRecord::Base.configurations.configs_for(env_name: "test", spec_name: spec_name)
+          ActiveRecord::Tasks::DatabaseTasks.load_schema(db_config, :ruby, filename)
+        ensure
+          if should_reconnect
+            ActiveRecord::Base.establish_connection(ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env, spec_name: "primary"))
+          end
+        end
+      end
+
+      # desc "Recreate the #{spec_name} test database from an existent structure.sql file"
+      namespace :load_structure do
+        task spec_name => "db:test:purge:#{spec_name}" do
+          filename = ActiveRecord::Tasks::DatabaseTasks.dump_filename(spec_name, :sql)
+          db_config = ActiveRecord::Base.configurations.configs_for(env_name: "test", spec_name: spec_name)
+          ActiveRecord::Tasks::DatabaseTasks.load_schema(db_config, :sql, filename)
+        end
+      end
+
+      # desc "Empty the #{spec_name} test database"
+      namespace :purge do
+        task spec_name => %w(load_config check_protected_environments) do
+          db_config = ActiveRecord::Base.configurations.configs_for(env_name: "test", spec_name: spec_name)
+          ActiveRecord::Tasks::DatabaseTasks.purge(db_config)
+        end
+      end
+
+      # desc 'Load the #{spec_name} database test schema'
+      namespace :prepare do
+        task spec_name => :load_config do
+          unless ActiveRecord::Base.configurations.blank?
```

Comment:
It may make sense to have a different check here or remove it entirely. The presence of the database might already indicate that the configuration exists.

Reply:
Yep, will do 👍